src/nvme/private.h

```diff
@@ -289,9 +305,11 @@ __u32 nvme_mi_crc32_update(__u32 crc, void *data, size_t len);
  * in the shared lib */;
 struct mctp_ioc_tag_ctl;
 struct __mi_mctp_socket_ops {
-	int (*socket)(int, int, int);
+	int (*msg_socket)(void);
+	int (*aem_socket)(__u8 eid, unsigned int network);
 	ssize_t (*sendmsg)(int, const struct msghdr *, int);
 	ssize_t (*recvmsg)(int, struct msghdr *, int);
+	ssize_t (*recv)(int __fd, void *__buf, size_t __n, int __flags);
```

Comment:
Are you sure we need this? `recvmsg` behaviour is a superset of `recv`

(granted, you do need to define a `struct msghdr` in the caller, but that should be less intrusive than a whole new callback)